Rework glossary (#2950)
theforeman/foreman-documentation: guides/common/modules/ref_glossary-of-terms.adoc

> [id="glossary-of-terms_{context}"]
> = Glossary of terms
> 
> This glossary documents various terms used in relation to {ProjectName}.

Reviewer: ```suggestion
[id="glossary-of-terms-used-in-project_{context}"]
= Glossary of terms used in {Project}
```
WDYT?

Author: Like it!

Author: 9001dc44f7b58386e5bf0ffd33953fd15b9047f0
